### PR TITLE
BUG: custom certSANs should append to the default ClusterConfiguration certSANs

### PR DESCRIPTION
**Pull request by hexinatgithub**
custom certSANs should append to the default ClusterConfiguration certSANs

### CODE REVIEW

**Comment by hexinatgithub** [comment]
#2688 
User may deploy an external LB to access the apiserver.

**Comment by fengxsong** [comment]
I think we just need to add one more mergo configuration to solve this problem

```
var defaultMergeOpts = []func(*mergo.Config){
	mergo.WithOverride,
	mergo.WithAppendSlice,
}
```

**Comment by hexinatgithub** [comment]
> I think we just need to add one more mergo configuration to solve this problem
> 
> ```
> var defaultMergeOpts = []func(*mergo.Config){
> 	mergo.WithOverride,
> 	mergo.WithAppendSlice,
> }
> ```
> 
> at the meantime, remove line https://github.com/labring/sealos/blob/main/pkg/runtime/kubeadm_config.go#L103
> 
> @cuisongliu needs some test.

And that's exactly what I thought in the beginning.
I'm not very sure other slice fields should append or overwrite it.

File: .github/workflows/test_run.yml [hunk 1]
```diff
@@ -348,6 +351,7 @@ jobs:
           sudo cat /root/.sealos/default/etc/kubeadm-init.yaml | grep systemd
           sudo cat /root/.sealos/default/etc/kubeadm-init.yaml | grep 100.64.0.0/10
           sudo cat /root/.sealos/default/etc/kubeadm-init.yaml | grep 100.55.0.0/16
+          sudo cat /root/.sealos/default/etc/kubeadm-init.yaml | grep 192.168.72.100
```
Comment (cuisongliu) [inline .github/workflows/test_run.yml:354]: need test all apiserver certSANs 

```
- 127.0.0.1
  - apiserver.cluster.local
  - 10.103.97.2
  - 10.1.0.127
  - 192.168.72.100
```

**Comment by cuisongliu** [comment]
> And that's exactly what I thought in the beginning.
> I'm not very sure other slice fields should append or overwrite it.
> Maybe some slice filed need be overwrite other than use default value?

see  https://github.com/labring/sealos/actions/runs/4362210104/jobs/7626881209#step:6:585 same test ok .

File: .github/workflows/test_run.yml (hunk 1 above)
Comment (hexinatgithub) [reply .github/workflows/test_run.yml:354]: Yes. Default value should include.

**Comment by hexinatgithub** [comment]
> > And that's exactly what I thought in the beginning.
> > I'm not very sure other slice fields should append or overwrite it.
> > Maybe some slice filed need be overwrite other than use default value?
> 
> see https://github.com/labring/sealos/actions/runs/4362210104/jobs/7626881209#step:6:585 same test ok .

Don't get it. What it mean?

**Comment by cuisongliu** [comment]
> Don't get it. What it mean?

https://github.com/labring/sealos/actions/runs/4362210104/jobs/7626881592#step:6:585

**Comment by hexinatgithub** [comment]
> > Don't get it. What it mean?
> 
> https://github.com/labring/sealos/actions/runs/4362210104/jobs/7626881592#step:6:585

https://github.com/labring/sealos/actions/runs/4362210104/workflow#L341 Test both Clusterfile and kubeadm-init.yaml?

**Comment by hexinatgithub** [comment]
> I think we just need to add one more mergo configuration to solve this problem
> 
> ```
> var defaultMergeOpts = []func(*mergo.Config){
> 	mergo.WithOverride,
> 	mergo.WithAppendSlice,
> }
> ```
> 
> at the meantime, remove line https://github.com/labring/sealos/blob/main/pkg/runtime/kubeadm_config.go#L103
> 
> @cuisongliu needs some test.

More conservative change will avoid introduce unnoticed bug.
@cuisongliu need some suggestion.

**Comment by cuisongliu** [comment]
> > > Don't get it. What it mean?
> > 
> > 
> > https://github.com/labring/sealos/actions/runs/4362210104/jobs/7626881592#step:6:585
> 
> https://github.com/labring/sealos/actions/runs/4362210104/workflow#L341 Test both Clusterfile and kubeadm-init.yaml?

Mainly to test kubeadm-init.yaml, it needs to cover more comprehensively, don’t lose the original configuration.

**Comment by cuisongliu** [comment]
> > I think we just need to add one more mergo configuration to solve this problem
> > ```
> > var defaultMergeOpts = []func(*mergo.Config){
> > 	mergo.WithOverride,
> > 	mergo.WithAppendSlice,
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > at the meantime, remove line https://github.com/labring/sealos/blob/main/pkg/runtime/kubeadm_config.go#L103
> > @cuisongliu needs some test.
> 
> More conservative change will avoid introduce unnoticed bug. @cuisongliu need some suggestion.

remove and test.

**Comment by fengxsong** [comment]
> I'm not very sure other slice fields should append or overwrite it.

one more concern, the [`scheme.Default(obj)`](https://github.com/labring/sealos/blob/main/pkg/runtime/kubeadm_config.go#L244)  should not invoked before complete merging configs.

**Comment by cuisongliu** [comment]
https://github.com/labring/sealos/blob/11c2f8fde2f6c0429771f6e4a36f1767a98d3d3c/.github/workflows/test_run.yml#L187  Also need to add test case

**Comment by hexinatgithub** [comment]
Done.

**Review by cuisongliu — APPROVED** [review]

**Pull request merged**